parsers/autoit.c

```diff
@@ -0,0 +1,77 @@
+/*
+*   Copyright (c) 2000-2006, Darren Hiebert, Elias Pschernig
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License version 2 or (at your option) any later version.
+*
+*   This module contains functions for generating tags for AutoIt functions.
+*/
+
+/*
+*   INCLUDE FILES
+*/
+#include "general.h"  /* must always come first */
+
+#include <string.h>
+
+#include "parse.h"
+#include "read.h"
+#include "routines.h"
+#include "vstring.h"
+
+/*
+*   DATA DEFINITIONS
+*/
+typedef enum {
+	K_FUNCTION
+} AutoItKind;
+
+static kindDefinition AutoItKinds[] = {
+	{ true, 'f', "func", "functions" }
+};
+
+/*
+*   FUNCTION DEFINITIONS
+*/
+static void findAutoItTags (void)
+{
+	vString *name = vStringNew ();
+	const unsigned char *line;
+
+	while ((line = readLineFromInputFile ()) != NULL)
+	{
+		const unsigned char* cp = line;
+		if ((line[0] == 'F' || line[0] == 'f') &&
+			(line[1] == 'U' || line[1] == 'u') &&
+			(line[2] == 'N' || line[2] == 'n') &&
+			(line[3] == 'C' || line[3] == 'c') &&
+			isspace ((int) line[4]))
+		{
+			cp += 4;
+			while (isspace ((int) *cp))
+				++cp;
+			while (isalnum ((int) *cp) || *cp == '_')
+			{
+				vStringPut (name, (int) *cp);
+				++cp;
+			}
+			while (isspace ((int) *cp))
+				++cp;
+			if (*cp == '(')
+				makeSimpleTag (name, K_FUNCTION);
```

Comment:
This parser is enough simple. So we can do the same thing with command line based regex parser:

```
[jet@localhost]~/var/ctags% cat /tmp/foo.au3 
; Taken from https://www.autoitscript.com/autoit3/docs/intro/lang_functions.htm

#include <Constants.au3>

Local $iNumber = 10
Local $iDoubled = 0

For $i = 1 To 10
    $iDoubled = MyDouble($iNumber)
    MsgBox($MB_OK, "", $iNumber & " doubled is " & $iDoubled)
    $iNumber = $iDoubled
Next
Exit

Func MyDouble($iValue)
    $iValue = $iValue * 2
    Return $iValue
EndFunc   ;==>MyDouble

func MyDouble0($iValue)
    $iValue = $iValue * 2
    Return $iValue
EndFunc   ;==>MyDouble

FUNC MyDouble1($iValue)
    $iValue = $iValue * 2
    Return $iValue
[jet@localhost]~/var/ctags% cat /tmp/autoit.ctags 
--langdef=AutoIt
--map-AutoIt=+.au3
--kinddef-AutoIt=f,funct,functions
--regex-AutoIt=/^func[[:space:]]{1,}([a-z0-9_]{1,}) *\(/\1/f/{icase}
[jet@localhost]~/var/ctags% ./ctags -o - --options=/tmp/autoit.ctags /tmp/foo.au3  
MyDouble	/tmp/foo.au3	/^Func MyDouble($iValue)$/;"	f
MyDouble0	/tmp/foo.au3	/^func MyDouble0($iValue)$/;"	f
MyDouble1	/tmp/foo.au3	/^FUNC MyDouble1($iValue)$/;"	f
```